### PR TITLE
tweaks test code for filter parser

### PR DESCRIPTION
**Pull request by fenglianxu**
Test code for #2298 
Signed-off-by: xufengli <xufengli@uk.ibm.com>

### CODE REVIEW

File: packages/loopback-connector-composer/test/filterparser.js [hunk 1]
```diff
@@ -308,17 +286,93 @@ describe('FilterParser', () => {
...
+        it('should return when a where condition with a between operator', () => {
+            (() => {
+                const keyObject = {'between':[1,10,20]};
+                FilterParser.parseObjectValue('f1', keyObject);
+            }).should.throw(/The between value is not an array or does not have two elements/);
+        });
+
     });
 
     describe('#isDateTime',()=> {
```
Comment (nklincoln) [inline packages/loopback-connector-composer/test/filterparser.js]: the test descriptions here don't match with what the tests actually do - the description says it should throw, but in fact it tests that they all return false?

File: packages/loopback-connector-composer/test/filterparser.js [hunk 2]
```diff
@@ -308,17 +286,93 @@ describe('FilterParser', () => {
...
+
+        it('should return when parse a number value', () => {
+            FilterParser.isPrimitiveTypeValue(1234567).should.be.true;
+        });
+
+        it('should return when parse a boolean value', () => {
+            FilterParser.isPrimitiveTypeValue(true).should.be.true;
+        });
+
+        it('should return when parse an Object value', () => {
```
Comment (nklincoln) [inline packages/loopback-connector-composer/test/filterparser.js]: should return false?

File: packages/loopback-connector-composer/test/filterparser.js [hunk 3]
```diff
@@ -308,17 +286,93 @@ describe('FilterParser', () => {
...
+            FilterParser.isDateTime(1234567).should.be.false;
+        });
+
+        it('should throw when parse an invalid datetime with an undefined value', () => {
+            FilterParser.isDateTime(undefined).should.be.false;
+        });
+
     });
 
     describe('#isPrimitiveTypeValue',()=> {
```
Comment (nklincoln) [inline packages/loopback-connector-composer/test/filterparser.js]: Need to be more descriptive in the descriptions here - the test should return something explicit, ie:
 -  'should return true when passed a value that is X '
 -  'should return false when passed a value that is Y'
 -  'should throw when passed a value that is Z'

File: packages/loopback-connector-composer/test/filterparser.js (hunk 2 above)
Comment (fenglianxu) [reply packages/loopback-connector-composer/test/filterparser.js]: updated the descriptions now

**Pull request merged**